fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfileParser.java

```diff
@@ -0,0 +1,330 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.starrocks.common.util;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.starrocks.thrift.TUnit;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.math.BigDecimal;
+import java.util.LinkedList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is used for reconstruct RuntimeProfile from plain text
+ */
+public class RuntimeProfileParser {
+    private static final Pattern BYTE_COUNTER_PATTERN =
+            Pattern.compile("^- (.*?): (-?\\d+\\.\\d+) (KB|MB|GB|TB|B)$");
+    private static final Pattern BYTE_PER_SEC_COUNTER_PATTERN =
+            Pattern.compile("^- (.*?): (-?\\d+\\.\\d+) (KB|MB|GB|TB|B)/sec$");
+    private static final Pattern UNIT_COUNTER_PATTERN =
+            Pattern.compile("^- (.*?): (-?\\d+(\\.\\d+)?([eE][-+]?\\d+)?)[BMK]?( \\((-?\\d+(\\.\\d+)?)\\))?$");
+    private static final Pattern UNIT_PER_SEC_COUNTER_PATTERN =
+            Pattern.compile("^- (.*?): (-?\\d+(\\.\\d+)?([eE][-+]?\\d+)?)[BMK]?( \\((-?\\d+(\\.\\d+)?)\\))? /sec$");
+    private static final Pattern TIMER_PATTERN =
+            Pattern.compile("^- (.*?): (((-?\\d+(\\.\\d+)?)(ms|us|ns|h|m|s))+)$");
+    private static final Pattern TIMER_SEGMENT_PATTERN =
+            Pattern.compile("(-?\\d+(\\.\\d+)?)(ms|us|ns|h|m|s)");
+    private static final Pattern INFO_STRING_PATTERN =
+            Pattern.compile("^- (.*?): (.*?)$");
+
+    public static RuntimeProfile parseFrom(String content) {
+        BufferedReader bufferedReader = new BufferedReader(new StringReader(content));
```

Comment:
Is this a totally new method? Has any unexpected problem may encounter when rebuilding profile from string?